ably/http/paginatedresult.py

```diff
@@ -40,35 +40,65 @@ def next(self):
     def __get_rel(self, rel_req):
         if rel_req is None:
             return None
-        return PaginatedResult.paginated_query_with_request(self.__http, rel_req, self.__response_processor)
+        return self.paginated_query_with_request(self.__http, rel_req, self.__response_processor)
 
-    @staticmethod
-    def paginated_query(http, url, headers, response_processor):
+    @classmethod
+    def paginated_query(cls, http, method='GET', url='/', body=None,
+                        headers=None, response_processor=None,
+                        raise_on_error=True):
         headers = headers or {}
-        req = Request(method='GET', url=url, headers=headers, body=None, skip_auth=False)
-        return PaginatedResult.paginated_query_with_request(http, req, response_processor)
-
-    @staticmethod
-    def paginated_query_with_request(http, request, response_processor):
-        response = http.request(request)
+        req = Request(method, url, body=body, headers=headers, skip_auth=False,
+                      raise_on_error=raise_on_error)
+        return cls.paginated_query_with_request(http, req, response_processor)
+
+    @classmethod
+    def paginated_query_with_request(cls, http, request, response_processor,
+                                     raise_on_error=True):
+        response = http.make_request(
+            request.method, request.url, headers=request.headers,
+            body=request.body, skip_auth=request.skip_auth,
+            raise_on_error=request.raise_on_error)
 
         items = response_processor(response)
 
         content_type = response.headers['Content-Type']
         links = response.links
-        log.debug("Links: %s" % links)
-        log.debug("Response: %s" % response)
+        #log.debug("Links: %s" % links)
```

Comment:
If we don't want these log entries, should we not just delete them?  It's always useful to allow a user to enable the most verbose log level to provide us with the complete request/response bodies & headers for debugging, but if this is not needed here then let's remove rather than comment.